specification/app/resource-manager/Microsoft.App/stable/2024-03-01/Diagnostics.json

```diff
@@ -0,0 +1,866 @@
+          },
+          {
+            "$ref": "./Jobs.json#/parameters/JobNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/DiagnosticsCollection"
+            }
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get the list of available diagnostic data for a Container App Job": {
+            "$ref": "./examples/Job_ListDetectors.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/jobs/{jobName}/detectors/{detectorName}": {
+      "get": {
+        "tags": [
+          "Jobs",
+          "Diagnostics"
+        ],
+        "summary": "Get the diagnostics data for a given Container App Job.",
+        "description": "Get the diagnostics data for a Container App Job.",
+        "operationId": "Jobs_GetDetector",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./Jobs.json#/parameters/JobNameParameter"
+          },
+          {
+            "name": "detectorName",
+            "in": "path",
+            "description": "Name of the Container App Job detector.",
+            "required": true,
+            "type": "string",
+            "pattern": "^[-\\w\\._\\(\\)]+$"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/Diagnostics"
+            }
+          },
+          "default": {
+            "description": "Common error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get diagnostic data for a Container App Job": {
+            "$ref": "./examples/Job_GetDetector.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/jobs/{jobName}/detectorProperties/{apiName}": {
```

Comment:
>detectorProperties/{apiName} [](http://example.com/codeflow?start=111&length=28)

The `apiName` seems a bit strange. Since it's after `detectorProperties`, I expect it be something like `propertyName`. The URI should follow the pattern: resourceCollection/resourceInstance.

Reply:
it is same with previous api version. I do not want to change it, which will cause breaking change.

Reply:
you have used the : "2023-05-01" as the base api version for this PR , and this api was not present there. 

Can you please clarify which previous api version are you talking about ? and why is that not used as the base commit for this PR ?

Reply:
We need to support all features in 2023-05-01, so we use it as base api version.
For the api you mentioned, it is added in https://github.com/Azure/azure-rest-api-specs/pull/27875 and the api is already added in previous api version 2023-08-01-preview https://github.com/Azure/azure-rest-api-specs/pull/26227